### PR TITLE
Readabilty + Documentation Tweaks && Add Index

### PR DESCRIPTION
**Pull request by zspecza**
- Adds support for `index` argument
- Fixes the link in the readme that links to my profile
- Adds better readability
- Adds better documentation

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,7 +8,7 @@ Map, reduce, forEach, and filter for objects. Lazy evaluation, no dependencies.
 
 I wanted a library that has no dependencies and gives me the basic map/reduce/filter for use on objects. Any existing library I found has boatloads of dependencies, provides tons more extra tools, and/or is unmaintained. So here's `ObjectFn`, just for you!
 
-Also, big props to [@declandewet](https://github.com/jescalan) for the initial implementation of this library!
+Also, big props to [@declandewet](https://github.com/declandewet) for the initial implementation of this library!
```
Comment (jescalan) [inline README.md:11]: Hahaha oh man honestly that was not on purpose hahah, must have mis-pasted

File: index.js [hunk 2]
```diff
@@ -1,38 +1,71 @@
-function * entries (obj) {
-  for (let key of Object.keys(obj)) {
-    yield [obj[key], key]
-  }
-}
+'use strict'
```
Comment (jescalan) [inline index.js]: I don't think you need this with node 6

File: README.md (hunk 1 above)
Comment (zspecza) [inline README.md:11]: I figured :P I'm not bothered tbh - this is what package.json `contributors` & `maintainers` and `contributors.md` & `maintainers.md` are for :)

File: index.js (hunk 2 above)
Comment (zspecza) [inline index.js]: Can't find any documentation/release notes stating so, but I still feel for the sake of setting a good example its better to have. There aren't any negative connotations to having it.

Comment (jescalan) [inline index.js]: I really don't like the extra boilerplate, and it doesn't have any obvious benefits for this particular app. There's [this](https://github.com/isaacs/use-strict) but still not great...

**Pull request merged**